incubator/hnc/internal/reconcilers/setup.go

```diff
@@ -9,25 +9,51 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 	"sigs.k8s.io/multi-tenancy/incubator/hnc/internal/forest"
 )
 
+type FakeDeleteCRDClient struct{}
+
+func (f FakeDeleteCRDClient) Get(context.Context, types.NamespacedName, runtime.Object) error {
+	return nil
+}
+
 var crds = map[string]bool{
 	"hncconfigurations.hnc.x-k8s.io":       true,
 	"subnamespaceanchors.hnc.x-k8s.io":     true,
 	"hierarchyconfigurations.hnc.x-k8s.io": true,
 }
 
+var deleteCRDClient interface {
+	Get(context.Context, types.NamespacedName, runtime.Object) error
+}
+
 // Create creates all reconcilers.
 //
 // This function is called both from main.go as well as from the integ tests.
-func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int) error {
+func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int, useFakeClient bool) error {
 	hcChan := make(chan event.GenericEvent)
 	anchorChan := make(chan event.GenericEvent)
 
+	// Create uncached client for CRD deletion check
+	if !useFakeClient {
+		// declaring err variable here to make sure we are using global variable deleteCRDClient later
+		var err error
+		deleteCRDClient, err = client.New(config.GetConfigOrDie(), client.Options{
+			Scheme: mgr.GetScheme(),
+			Mapper: mgr.GetRESTMapper(),
```

Comment:
We weren't sure if this was actually needed - maybe add a comment that says we don't know, or maybe test taking this out and see if it still works :D (though that might be overkill).